science/gnuradio/Portfile

```diff
@@ -462,61 +321,58 @@ variant docs description "Install GNU Radio documentation" {
         port:xmlto
 
     depends_build-append \
-        port:py${PythonVersionNoDot}-sphinx \
+        port:py${active_python_version_no_dot}-sphinx \
         port:texlive-latex
 
     configure.args-append \
         -DENABLE_DOXYGEN=ON \
         -DENABLE_SPHINX=ON \
-        -DSPHINX_EXECUTABLE=${frameworks_dir}/Python.framework/Versions/${PythonVersionWithDot}/bin/sphinx-build
-
-}
-
-if {![variant_isset docs]} {
-
-    configure.args-append \
-        -DENABLE_DOXYGEN=OFF \
-        -DENABLE_SPHINX=OFF
+        -DSPHINX_EXECUTABLE=${python_framework_dir}/bin/sphinx-build
```

Comment:
since some of these cmake variables are set above, you'll need to do a "args-replace" here ... and, anywhere the above defaults are to be changed. hence why my prior methods of not setting them until the variant selection is known ... and you can always go back to that method if you like :)

Reply:
cmake takes always the last; anyway, you are right, we should declare the delete/replace.